### PR TITLE
Playback 2024: Ratings

### LINKED ISSUE #2353: Ratings Chart screen

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: #2250 |
|:---:|

Fixes #2353

| | |
|--|--|
| <video src="https://github.com/user-attachments/assets/f1b361b4-a862-493f-bb5b-9a3df02ec268"> | <video src="https://github.com/user-attachments/assets/2962f9e4-6816-4259-9525-03eb2c670885"> |

| | |
|--|--|
| <video src="https://github.com/user-attachments/assets/8e86f670-8ca4-457c-9c86-023948923785"> | <video src="https://github.com/user-attachments/assets/22ba38a2-aad3-4a09-b87c-38ca616f20a2"> |

### Loading Ratings
* Loads Ratings using the API endpoint
* Caches loaded ratings on DataManager object
* Adds check for loaded ratings in when loading story

### Ratings Screen
* Adds Rating Chart
* Adds Empty view when there are no ratings with Learn More button
* Adds SafariView to present load button and handles dismissal to pause & then resume playback on the story

### Ratings Chart

* Use an account which has left several ratings
* Enable `endOfYear2024`
* Restart app
* Go to Profile and tap "Playback 2024"
* Navigate until you get to the Ratings screen

### Ratings Chart

* Use an account which has never left ratings
* Enable `endOfYear2024`
* Restart app
* Go to Profile and tap "Playback 2024"
* Navigate until you get to the Ratings screen
* Ensure that the "Oopsies" screen shows
* Tap the "Learn about ratings" button
* Verify that the web view shows and that sitting here for 5+ seconds doesn't continue the story progression in the background
* Dismiss web view and verify that story resumes

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/End of Year/End of Year 2024/EndOfYear2024StoriesModel.swift [hunk 1]
```diff
@@ -94,11 +102,17 @@ class EndOfYear2024StoriesModel: StoryModel {
...
+        case .ratings:
+            return true
         default:
             return false
         }
     }
 
+    func hasLoadedData(in dataManager: DataManager) -> Bool {
+        dataManager.ratings.ratings == nil
+    }
```
Comment (danielebogo) [inline podcasts/End of Year/End of Year 2024/EndOfYear2024StoriesModel.swift]: This is a bit confusing. If I call `hasLoadedData` I'm not expecting `ratings` to be nil. Can we find a better name? If it's only related to ratings I would specify that in the function name.

**Review by danielebogo — CHANGES_REQUESTED** [review]
Hey @bjtitus I get a loading error while runnig this branch. I think something happens when calling the `EndOfYearStoriesBuilder.build()`.
I also left a comment about one function name.

**Review by danielebogo — APPROVED** [review]
Thanks Brandon! Now it's working fine!

**Pull request merged**